cmake/FDBComponents.cmake

```diff
@@ -20,41 +20,36 @@ endif()
 
 include(CheckSymbolExists)
 
-set(DISABLE_TLS OFF CACHE BOOL "Don't try to find OpenSSL and always build without TLS support")
 set(USE_WOLFSSL OFF CACHE BOOL "Build against WolfSSL instead of OpenSSL")
 set(USE_OPENSSL ON CACHE BOOL "Build against OpenSSL")
-if(DISABLE_TLS)
-  set(WITH_TLS OFF)
-else()
-  if(USE_WOLFSSL)
-    set(WOLFSSL_USE_STATIC_LIBS TRUE)
-    find_package(WolfSSL)
-    if(WOLFSSL_FOUND)
-      set(CMAKE_REQUIRED_INCLUDES ${WOLFSSL_INCLUDE_DIR})
-      set(WITH_TLS ON)
-      add_compile_options(-DHAVE_OPENSSL)
-      add_compile_options(-DHAVE_WOLFSSL)
-    else()
-      message(STATUS "WolfSSL was not found - Will compile without TLS Support")
-      message(STATUS "You can set WOLFSSL_ROOT_DIR to help cmake find it")
-      set(WITH_TLS OFF)
-    endif()
-  elseif(USE_OPENSSL)
-    set(OPENSSL_USE_STATIC_LIBS TRUE)
-    if(WIN32)
-      set(OPENSSL_MSVC_STATIC_RT ON)
-    endif()
-    find_package(OpenSSL)
-    if(OPENSSL_FOUND)
-      set(CMAKE_REQUIRED_INCLUDES ${OPENSSL_INCLUDE_DIR})
-      set(WITH_TLS ON)
-      add_compile_options(-DHAVE_OPENSSL)
-    else()
-      message(STATUS "OpenSSL was not found - Will compile without TLS Support")
-      message(STATUS "You can set OPENSSL_ROOT_DIR to help cmake find it")
-      set(WITH_TLS OFF)
-    endif()
+if(USE_WOLFSSL)
+  set(WOLFSSL_USE_STATIC_LIBS TRUE)
+  find_package(WolfSSL)
+  if(WOLFSSL_FOUND)
+    set(CMAKE_REQUIRED_INCLUDES ${WOLFSSL_INCLUDE_DIR})
+    add_compile_options(-DHAVE_OPENSSL)
+    add_compile_options(-DHAVE_WOLFSSL)
+  else()
+    message(STATUS "WolfSSL was not found - Will compile without TLS Support")
+    message(STATUS "You can set WOLFSSL_ROOT_DIR to help cmake find it")
+    message(FATAL_ERROR "Unable to find WolfSSL")
+  endif()
+elseif(USE_OPENSSL)
+  set(OPENSSL_USE_STATIC_LIBS TRUE)
+  if(WIN32)
+    set(OPENSSL_MSVC_STATIC_RT ON)
   endif()
+  find_package(OpenSSL)
```

Comment:
I am not sure whether you want to do this in this PR, but we should prettify this now. So instead of the if-else block below, you could just do `find_package(OpenSSL REQUIRED)`. Then, instead of changing `CMAKE_REQUIRED_INCLUDES` it would be cleaner to link against the right OpenSSL targets